_docs/getting-started/mac.md

```diff
@@ -6,22 +6,44 @@ The Mac build works easiest with Anaconda. Always pull the latest from github, s
 
 ```bash
 git clone --recursive https://github.com/caffe2/caffe2.git && cd caffe2
 conda build conda
 conda install caffe2 --use-local
 ```
 
-This will build caffe2 using [conda build](https://conda.io/docs/user-guide/tasks/build-packages/recipe.html), with the flags specified in `conda/build.sh` and the packages specified in `conda/meta.yaml`. `conda build` will create a conda package (tarball) on your machine, which `conda install` then installs. To build caffe2 with different settings, change the dependencies in `meta.yaml` and the `CMAKE_ARGS` flags in `conda/build.sh` and run the above command again. 
+This will build caffe2 using [conda build](https://conda.io/docs/user-guide/tasks/build-packages/recipe.html), with the flags specified in `conda/build.sh` and the packages specified in `conda/meta.yaml`. `conda build` will create a conda package (tarball) on your machine, which `conda install` then installs. To build caffe2 with different settings, change the dependencies in `meta.yaml` and the `CMAKE_ARGS` flags in `conda/build.sh` and run the above commands again. 
 
 If your default Anaconda Python is not 2.7, you can install a different version of Python using `conda create --name python2 python=2` (`python2` can be any name you like.)  Subsequently, if you `source activate python2`, your path will be adjusted so that you get `python2`. See [this page](https://conda.io/docs/user-guide/tasks/manage-environments.html) on managing conda environments for more info.
 
 ### Optional GPU Support
 
 In the instance that you have a NVIDIA supported GPU in your Mac, then you should visit the NVIDIA website for [CUDA](https://developer.nvidia.com/cuda-downloads) and [cuDNN](https://developer.nvidia.com/cudnn) and install the provided binaries. You will have to remove `-DUSE_CUDA=OFF` and `-DUSE_NCCL=OFF` flags from `conda/build.sh`.
 
+### Custom Anaconda Install
+
+If you plan to change the source code of caffe2 or want to customize the build process, you may want to bypass conda and call cmake manually. The following commands will build caffe2 in a directory called `build` under your caffe2 root and install caffe2 in a conda env. In this example Anaconda is installed in `~/anaconda2`.
+
+```bash
+# Create a conda environment (optional)
+conda create -n my_caffe2_env && source activate my_caffe2_env
+
+# Install the packages you want to build caffe2 with here
+# e.g.
+# conda install protobuf mkl mkl-include
+
+# Clone caffe2
+cd ~ && git clone --recursive https://github.com/caffe2/caffe2.git && cd caffe2
+
+# Make caffe2 in a separate folder
+mkdir build && cd build
+cmake <other flags here> -DCMAKE_PREFIX_PATH=~/anaconda2/envs/my_caffe2_env -DCMAKE_INSTALL_PREFIX=~/anaconda2/envs/my_caffe2_env ..
```

Comment:
nit: may be remove this <other flags here> so its copypastable.

Reply:
Okay I think I'll add in installing all the packages too so that tutorials work out of the box